src/interrupts.rs

```diff
@@ -0,0 +1,41 @@
+use crate::*;
+
+// NOTE: ESP-IDF-specific
+const SPINLOCK_FREE_INIT: u32 = 0xB33FFFFF;
+pub type EspCriticalMutex = portMUX_TYPE;
```

Comment:
As mentioned in the general comment, this code does not compile on single-core ESP MCUs.